libcodechecker/server/routing.py

```diff
@@ -47,18 +47,45 @@ def is_valid_product_endpoint(uripart):
     return True
 
 
-def get_product_name(path):
+def split_client_GET_request(path):
     """
-    Get product name from the request's URI.
+    Split the given request URI to its parts relevant to the server.
+
+    Returns the product endpoint and the "remainder" of the request path
+    as a tuple of 2.
     """
 
-    # A standard request from a browser looks like:
+    # A standard GET request from a browser looks like:
     # http://localhost:8001/[product-name]/#{request-parts}
     # where the parts are, e.g.: run=[run_id]&report=[report_id]
-    #
-    # Rewrite the "product-name" so that the web-server deploys the
-    # viewer client from the www/ folder.
 
-    # The split array looks like ['', 'product-name', ...].
-    first_part = urlparse.urlparse(path).path.split('/', 2)[1]
-    return first_part if is_valid_product_endpoint(first_part) else None
+    parsed_path = urlparse.urlparse(path).path
+    split_path = parsed_path.split('/', 2)
```

Comment:
I do not like magic numbers `2`, `3` in the code `split_path[1]` .... Couldn't we use a named tuple or something else which is less error prone?

Reply:
I could introduce variable names to the return statements, but eventually these variables will also be introduced from an index in the split list.